python/cuml/cuml/manifold/umap.pyx

```diff
@@ -766,19 +775,33 @@ class UMAP(Base,
         # Handle dense inputs
         else:
             self._sparse_data = False
-            if data_on_host:
+            # Get nnd_n_clusters value for validation
+            build_kwds = self.build_kwds or {}
+            nnd_n_clusters = build_kwds.get("nnd_n_clusters", 1)
```

Comment:
I _believe_ the user can still have `nnd_n_clusters` in `build_kwds` even if `build_algo="brute_force_knn"`.

If so, we should include `build_algo` in the logic here as well. Perhaps something like:

```
if self.build_algo == "brute_force_knn":
    streaming = False
else:
    streaming = build_kwds.get("nnd_n_clusters", 1) > 1
```

then branch on `streaming` below rather than `nnd_n_clusters > 1`.

Reply:
Aha yes makes sense thanks for pointing out!